### PR TITLE
Code smells final

### PR DESCRIPTION
**Pull request by FelixDaCat7**
Some code smell resolutions for sprint 4

### CODE REVIEW

File: app/src/main/java/com/conupods/OutdoorMaps/Services/OutdoorDirectionsService.java [hunk 1]
```diff
@@ -107,7 +100,7 @@ private void setDirectionRoutes(DirectionsRoute[] routes) {
         if (this.mDirectionRoutes != null) {
             return mDirectionRoutes;
         } else {
-            return mDirectionRoutes;
+            return null;
```
Comment (finndavidson) [inline app/src/main/java/com/conupods/OutdoorMaps/Services/OutdoorDirectionsService.java:103]: doesnt this mean its unnecessary

Comment (FelixDaCat7) [reply app/src/main/java/com/conupods/OutdoorMaps/Services/OutdoorDirectionsService.java:103]: yeah i i could indeed remove the whole if, ill wait to see if there other comments and ill push this later

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java [hunk 2]
```diff
@@ -2,20 +2,20 @@
 
 public class Photo
 {
-    private String photo_reference;
+    private String photoReference;
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java:5]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java [hunk 3]
```diff
@@ -2,20 +2,20 @@
...
 
     private String width;
 
     private String height;
 
-    public String getPhoto_reference ()
+    public String getPhotoReference()
     {
-        return photo_reference;
+        return photoReference;
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java:13]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java [hunk 4]
```diff
@@ -2,20 +2,20 @@
...
 {
-    private String photo_reference;
+    private String photoReference;
 
     private String width;
 
     private String height;
 
-    public String getPhoto_reference ()
+    public String getPhotoReference()
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java:11]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java [hunk 5]
```diff
@@ -2,20 +2,20 @@
...
     {
-        return photo_reference;
+        return photoReference;
     }
 
-    public void setPhoto_reference (String photo_reference)
+    public void setPhotoReference(String photoReference)
     {
-        this.photo_reference = photo_reference;
+        this.photoReference = photoReference;
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java:18]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java [hunk 6]
```diff
@@ -2,20 +2,20 @@
...
 
-    public String getPhoto_reference ()
+    public String getPhotoReference()
     {
-        return photo_reference;
+        return photoReference;
     }
 
-    public void setPhoto_reference (String photo_reference)
+    public void setPhotoReference(String photoReference)
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java:16]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java [hunk 7]
```diff
@@ -42,7 +42,7 @@ public void setHeight (String height)
     @Override
     public String toString()
     {
-        return "ClassPojo [photo_reference = "+photo_reference+", width = "+width+", html_attributions = "+", height = "+height+"]";
+        return "ClassPojo [photo_reference = "+ photoReference +", width = "+width+", html_attributions = "+", height = "+height+"]";
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Photo.java:45]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Services/PlacesService.java [hunk 8]
```diff
@@ -118,16 +118,10 @@ private void requestPOIs(LatLng requestLatLng, String POICriteria) {
...
                                         Place place = response.body().getResults()[i];
-                                        double latitude = Double.parseDouble(place.getGeometry().getLocation().getLat());
-                                        double longitude = Double.parseDouble(place.getGeometry().getLocation().getLng());
-
-                                        LatLng latLng = new LatLng(latitude, longitude);
-                                        String placeName = place.getName();
 
                                         if (place.getPhotos() != null) {
-                                            String photoReference = place.getPhotos()[0].getPhoto_reference();
+                                            String photoReference = place.getPhotos()[0].getPhotoReference();
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Services/PlacesService.java:124]: same, thing, revert change to match POJO

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Place.java [hunk 9]
```diff
@@ -10,7 +10,7 @@
 
     private String id;
 
-    private String place_id;
+    private String placeId;
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Place.java:13]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Place.java [hunk 10]
```diff
@@ -65,19 +65,19 @@ public void setId (String id)
         this.id = id;
     }
 
-    public String getPlace_id ()
+    public String getPlaceId()
     {
-        return place_id;
+        return placeId;
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Place.java:70]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Place.java [hunk 11]
```diff
@@ -65,19 +65,19 @@ public void setId (String id)
...
     {
-        this.place_id = place_id;
+        this.placeId = placeId;
     }
 
     @Override
     public String toString()
     {
-        return "ClassPojo [types = "+", icon = "+", rating = "+", photos = "+photos+", reference = "+", user_ratings_total = "+", scope = "+", name = "+name+", opening_hours = "+", geometry = "+", vicinity = "+", id = "+id+", plus_code = "+", place_id = "+place_id+"]";
+        return "ClassPojo [types = "+", icon = "+", rating = "+", photos = "+photos+", reference = "+", user_ratings_total = "+", scope = "+", name = "+name+", opening_hours = "+", geometry = "+", vicinity = "+", id = "+id+", plus_code = "+", place_id = "+ placeId +"]";
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/Place.java:81]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java [hunk 12]
```diff
@@ -1,46 +1,38 @@
...
-        return html_attributions;
+    public String[] gethtmlAttributions() {
+        return htmlAttributions;
     }
 
-    public void setHtml_attributions (String[] html_attributions)
-    {
-        this.html_attributions = html_attributions;
+    public void setHtmlAttributions(String[] htmlAttributions) {
+        this.htmlAttributions = htmlAttributions;
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java:15]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java [hunk 13]
```diff
@@ -1,46 +1,38 @@
 package com.conupods.OutdoorMaps.Models.PointsOfInterest;
 
-public class PlacesOfInterest
-{
-    private String[] html_attributions;
+public class PlacesOfInterest {
+    private String[] htmlAttributions;
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java:4]: Do not change the name of this attribute, it must match the attribute name that is within the JSON response object. Revert change.

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java [hunk 14]
```diff
@@ -1,46 +1,38 @@
...
 
-    public void setResults (Place[] results)
-    {
+    public void setResults(Place[] results) {
         this.results = results;
     }
 
-    public String getStatus ()
-    {
+    public String getStatus() {
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java:26]: same thing

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java [hunk 15]
```diff
@@ -1,46 +1,38 @@
...
+    public void setStatus(String status) {
         this.status = status;
     }
 
     @Override
-    public String toString()
-    {
-        return "ClassPojo [html_attributions = "+html_attributions+", results = "+results+", status = "+status+"]";
+    public String toString() {
+        return "ClassPojo [html_attributions = " + htmlAttributions + ", results = " + results + ", status = " + status + "]";
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java:36]: same thing

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java [hunk 16]
```diff
@@ -1,46 +1,38 @@
...
 
-    public String getStatus ()
-    {
+    public String getStatus() {
         return status;
     }
 
-    public void setStatus (String status)
-    {
+    public void setStatus(String status) {
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java:30]: same thing

File: app/src/main/java/com/conupods/OutdoorMaps/View/Directions/NavigationActivity.java [hunk 17]
```diff
@@ -213,25 +200,22 @@ public void onFailure(Throwable e) {
...
+            if (mDestinationLongName != null && mDestinationCode != null) {
+                requestIndoorDirections();
             }
         });
     }
 
     private void requestIndoorDirections() {
         ArrayList<Spot> endPoints = new ArrayList<>();
-        mIndoorPath = new IndoorPath();
+        IndoorPath mIndoorPath = new IndoorPath();
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/View/Directions/NavigationActivity.java:218]: no longer a member variable, remove "m"

File: app/src/main/java/com/conupods/OutdoorMaps/View/Directions/NavigationActivity.java [hunk 18]
```diff
@@ -273,24 +257,19 @@ private void requestIndoorDirections() {
 
     // polyline function - can be modified to work with a single route
     private void addPolyLinesToMap(final DirectionsResult result) {
-        new Handler(Looper.getMainLooper()).post(new Runnable() {
-            @Override
-            public void run() {
-                for (DirectionsRoute route : result.routes) {
-                    List<com.google.maps.model.LatLng> decodedPath = PolylineEncoding.decode(route.overviewPolyline.getEncodedPath());
+        new Handler(Looper.getMainLooper()).post(() -> {
+            for (DirectionsRoute route : result.routes) {
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/View/Directions/NavigationActivity.java:261]: the logic to lay polylines needs to be in the run method

File: app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java [hunk 19]
```diff
@@ -1,46 +1,38 @@
...
 
-    public void setResults (Place[] results)
-    {
+    public void setResults(Place[] results) {
         this.results = results;
     }
 
-    public String getStatus ()
-    {
+    public String getStatus() {
```
Comment (mcalixte) [inline app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java:26]: still needs to change

Comment (FelixDaCat7) [reply app/src/main/java/com/conupods/OutdoorMaps/Models/PointsOfInterest/PlacesOfInterest.java:26]: not sure what there is to change here

File: app/src/main/java/com/conupods/OutdoorMaps/View/Directions/NavigationActivity.java (hunk 18 above)
Comment (minazenhub) [reply app/src/main/java/com/conupods/OutdoorMaps/View/Directions/NavigationActivity.java:262]: thats the lambda function this should be fine

**Comment by FelixDaCat7** [comment]
Arisen bugs from these commits make this PR too risky to merge into master this close from the final demo.

**Pull request closed without merging**